### PR TITLE
Feature/todo

### PR DESCRIPTION
**Pull request by hatim523**

### CODE REVIEW

File: src/api/helpers.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import axios from "axios";
+import { APIHost } from "../db/settings";
+
+
+function addNewTodo(newTodo) {
```
Comment (iamalisajid) [inline src/api/helpers.js]: use es6 arrow functions also create separate API component which will initialize axios only once.

File: src/api/helpers.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import axios from "axios";
+import { APIHost } from "../db/settings";
```
Comment (iamalisajid) [inline src/api/helpers.js]: This should be BASE_URL and it shouldn't be coming from db/settings. 
Move this to .env

File: src/api/helpers.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+function addNewTodo(newTodo) {
+    return axios.post(`${APIHost}todos`, newTodo);
+}
+
+function removeTodo(todoId) {
+    return axios.delete(`${APIHost}todos/${todoId}`);
+}
+
+function toggleComplete(todoId, isCompleted) {
+    return axios.patch(`${APIHost}todos/${todoId}`, {complete: isCompleted})
```
Comment (iamalisajid) [inline src/api/helpers.js]: Indent it properly use eslint with prettier for formatting.

File: src/api/helpers.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+}
+
+export {addNewTodo, removeTodo, toggleComplete};
```
Comment (iamalisajid) [inline src/api/helpers.js]: Add EOF

File: src/components/App.js [hunk 5]
```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+import axios from 'axios'
+import TodoItem from './todoItems';
+import AddTodo from './addTodo'
+import { APIHost } from '../db/settings';
+import { addNewTodo, removeTodo, toggleComplete } from '../api/helpers';
+
+export default class App extends React.Component {
```
Comment (iamalisajid) [inline src/components/App.js]: No need of extending it from Component also use arrow functions

Comment (iamalisajid) [inline src/components/App.js]: Do the export default in the end of the file.

File: src/components/App.js [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+    constructor(props) {
+        super(props);
+        this.state = {
```
Comment (iamalisajid) [inline src/components/App.js]: Use hooks for managing the state.

File: src/components/App.js [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+            items: [],
+        };
+    }
+
+    componentDidMount() {
```
Comment (iamalisajid) [inline src/components/App.js]: Don't use these life cycle methods learn about hooks and modern react and use it.

File: src/styles/todoItems.css [hunk 8]
```diff
@@ -0,0 +1,52 @@
+
+.todo-item {
```
Comment (iamalisajid) [inline src/styles/todoItems.css]: use module.scss and ad it next to the component file

File: src/styles/todoItems.css [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+    background-color: #4CAF50;
+}
+
+.todo-text {
+    padding: 1rem;
+    color: black;
+}
+
+.button {
+    background-color: #4CAF50; /* Green */
```
Comment (iamalisajid) [inline src/styles/todoItems.css]: sort all the properties alphabetically and keep the colors in a separate file and import it.

**Review by iamalisajid — CHANGES_REQUESTED** [review]
learn about react hooks see some tutorials of new version and implement it accordingly.

File: src/styles/todoItems.css (hunk 9 above)
Comment (hatim523) [reply src/styles/todoItems.css]: Sorted properties alphabetically

File: src/api/api.js [hunk 10]
```diff
@@ -0,0 +1,5 @@
+import axios from 'axios';
+
+export default axios.create({
+  baseURL: process.env.REACT_APP_DB_URL,
```
Comment (iamalisajid) [inline src/api/api.js]: REACT_APP_DB_ROOT_URL

File: src/styles/colors.scss [hunk 11]
```diff
@@ -0,0 +1,5 @@
+
+$complete-color: #4CAF50;
+$black-color: black;
+$white-color: white;
+$red-color: #f44336;
```
Comment (iamalisajid) [inline src/styles/colors.scss]: eof

**Review by iamalisajid — CHANGES_REQUESTED** [review]

**Pull request merged**